src/main.rs

```diff
@@ -229,3 +289,16 @@ fn ram_x() {
 fn ram_y() {
     dbg!(ram_y as fn() as *const ());
 }
+
+#[link_section = ".data"]
+#[interrupt]
+fn SIO_IRQ_PROC0() {
```

Comment:
I think in total, we use 2 interrupts. This one and the DMA one. To me, it is not super clear when these are being in the scanline process and where they fit in the overall architecture.

Reply:
Right. The DMA interrupt is called (on core 1) every scan line, like clockwork. This is is called by the system only when core 1 is overloaded and can't handle all the rendering work, and requests core 0 to render one scan line worth of content.